src/blocks/folder-blocks/infinite-canvas/Item.tsx

```diff
@@ -0,0 +1,163 @@
+import { useMemo, useRef, useState } from "react";
+import { useDrag } from "./useDrag";
+import { BlockPicker } from "./BlockPicker";
+import { ResizeButton } from "./ResizeButton";
+import { roundToInterval, Position, ItemType, Dimensions } from "./index";
+
+export const Item = ({
+  type,
+  text,
+  path,
+  url,
+  block,
+  blockProps,
+  contents = "",
+  position,
+  dimensions,
+  blockOptions = [],
+  BlockComponent,
+  onDelete,
+  onChange,
+  onDrag,
+}: ItemType & {
+  contents?: string;
+  blockOptions: any[];
+  BlockComponent?: any;
+  blockProps?: any;
+  onDelete: () => void;
+  onChange: (newContents: Partial<ItemType>) => void;
+}) => {
+  const [textBuffer, setTextBuffer] = useState(text);
+  const headerElement = useRef<HTMLDivElement>(null);
+
+  const { isDragging } = useDrag(
+    headerElement,
+    position,
+    (newDimensions: Dimensions) => {
+      onChange({
+        position: newDimensions.map((d) =>
+          roundToInterval(d, 10)
+        ) as Dimensions,
+      });
+    }
+  );
+
+  const relevantBlockOptions = useMemo(() => {
+    if (type !== "file") return null;
+    const extension = path?.split(".").pop();
+    return blockOptions.filter((block: any) => {
+      if (block.type !== "file") return false;
+      if (!block.extensions) return true;
+      if (block.extensions?.includes("*")) return true;
+      if (block.extensions?.includes(extension)) return true;
+      return false;
+    });
+  }, [blockOptions, path, type]);
+
+  const headerColor = {
+    text: "bg-blue-50",
+    file: "bg-red-50",
+  }[type];
+  const zIndex = {
+    text: "z-30",
+    file: "z-10",
+  }[type];
+
+  return (
+    <div
+      className={`item group Box shadow ${zIndex}`}
+      style={{
+        width: dimensions[0],
+        height: dimensions[1],
+        transform: `translate(${position[0]}px, ${position[1]}px)`,
+      }}
+    >
+      {/* grabby header */}
+      <div
+        className={`${headerColor} h-full w-full flex-none`}
+        style={{
+          cursor: isDragging ? "grabbing" : "grab",
+        }}
+        ref={headerElement}
+      />
+
+      <div className="px-3 py-2 w-full overflow-auto">
+        {type === "text" ? (
+          <div
+            className="Text focus:outline-none cursor-text w-full h-full text-sm"
+            contentEditable
```

Comment:
hmmm not sure if this is important, it is very stern
```
Warning: A component is `contentEditable` and contains `children` managed by React. It is now your responsibility to guarantee that none of those nodes are unexpectedly modified or duplicated. This is probably not intentional.
```

Reply:
it _is_ very stern! We're okay here, not worried about the contents being blown away because we're catching any changes with `onChange`. TIL about `suppressContentEditableWarning`, though, which I'll throw on there to prevent those warnings in the future